### PR TITLE
Added N-Queen Visualizer | Issue #825

### PR DESCRIPTION
**Pull request by Shariq2003**
## Description 

**N-Queens Visualiser**! This web application allows you to visualize the classic N-Queens problem, where the objective is to place N queens on an N x N chessboard such that no two queens threaten each other. This is achieved by ensuring that no two queens share the same row, column, or diagonal.

Closes: #825 

- **Title:**  N-Queen Visualizer
- **Name:** Shariq
- **Idenitfy yourself:** 
GSSOC Ext

### Describe the add-ons or changes you've made 📃
Added N-Queen Visualizer, a project of Backtracking that help to understand the N-Queen Problem by the Viusualization of it.

### Screenshots 📷

image
image

https://github.com/user-attachments/assets/052766a2-0488-4071-989a-54699e09dfd0

### Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist: ☑️

- [X] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly wherever it was hard to understand.
- [X] My changes generate no new warnings.

### CODE REVIEW

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney sir, please review

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney sir, please review this PR,

File: Vanilla-JS-Projects/Advanced/N-Queen-Visualizer/README.md
Comment (Avdhesh-Varshney) [inline Vanilla-JS-Projects/Advanced/N-Queen-Visualizer/README.md:1]: Follow the project README template.

File: Vanilla-JS-Projects/README.md
Comment (Avdhesh-Varshney) [inline Vanilla-JS-Projects/README.md:1]: image
Add your project in row 4 not create row 5

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney please check

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney sir ,please review now

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney sir, now i have reviewed all the changes and make the PR ready to merge.
Please review

**Comment by Shariq2003** [comment]
@Avdhesh-Varshney sir, could u please merge the PR in order they are creating in time, as I have creates this PR which has no conflicts now, but if u merge a PR that has raise after me, then there will be conflicts in my PR in readme.md file.

So please merge this

**Review by Avdhesh-Varshney — APPROVED** [review]
This time, I've done changes. Review them by your own so that next time. It will not repeat.

**Comment by NEGRAH-S** [comment]
@MastanSayyad , Only Project addition cannot allow Level 3

**Review by MastanSayyad — COMMENTED** [review]
@Avdhesh-Varshney reduce to `level 2`

**Comment by Shariq2003** [comment]
> @Avdhesh-Varshney reduce to `level 2`

Agreed

**Pull request merged**